ec/src/scalar_mul/glv.rs

```diff
@@ -29,9 +31,8 @@ pub trait GLVConfig: Send + Sync + 'static + SWCurveConfig {
     ) -> ((bool, Self::ScalarField), (bool, Self::ScalarField)) {
         let scalar: BigInt = k.into_bigint().into().into();
 
-        let coeff_bigints: [BigInt; 4] = Self::SCALAR_DECOMP_COEFFS.map(|x| {
-            BigInt::from_biguint(x.0.then_some(Sign::Plus).unwrap_or(Sign::Minus), x.1.into())
-        });
+        let coeff_bigints: [BigInt; 4] = Self::SCALAR_DECOMP_COEFFS
+            .map(|x| BigInt::from_biguint(if x.0 { Sign::Plus } else { Sign::Minus }, x.1.into()));
```

Comment:
Maybe we should just extract the sign into a separate variable? It's a bit noisy atm

Reply:
fixed